gtdynamics/factors/CollocationFactors.h

```diff
@@ -76,10 +76,10 @@ class EulerPoseCollocationFactor
   gtsam::Vector evaluateError(
       const gtsam::Pose3 &pose_t0, const gtsam::Pose3 &pose_t1,
       const gtsam::Vector6 &twist, const double &dt,
-      boost::optional<gtsam::Matrix &> H_pose_t0 = boost::none,
-      boost::optional<gtsam::Matrix &> H_pose_t1 = boost::none,
-      boost::optional<gtsam::Matrix &> H_twist = boost::none,
-      boost::optional<gtsam::Matrix &> H_dt = boost::none) const override {
+      gtsam::OptionalMatrixType H_pose_t0 = nullptr,
```

Comment:
this is an instance where, for consistency, it might make sense to replace all boost::none with `nullptr` or `std::nullopt_t` instead of {}

Reply:
I'm following the same defaults as GTSAM for consistency.